pkg/pillar/cmd/upgradeconverter/convertglobalconfig.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package upgradeconverter
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+)
+
+func handleUpgradeGlobalConfig(action ConverterAction, oldFile string, newFile string) (err error) {
+	switch action {
+	case Convert:
+		err = convert(oldFile, newFile)
+	case CleanupNew:
+		err = cleanupNew(oldFile, newFile)
+	case CleanupOld:
+		err = cleanupOld(oldFile)
+	case NoAction:
+		break
+	}
+	return err
+}
+
+func convert(oldVersionPath string, newVersionPath string) (err error) {
+	oldVersionExists := fileExists(oldVersionPath)
+	newVersionExists := fileExists(newVersionPath)
```

Comment:
To make the decisions you need two pieces of information:
1. the file (old or new) exists, can be read, and can be json decoded
2. the timestamp on the file

I think the logic will be simpler if you determine all at once with a 
getOld(oldVersionPath) which returns a bool exists, a timestamp, and types.GlobalConfig
and similarly for getNew()
Then you don't have to handle errors like can't read or decode later.